### ISSUE TITLE
Manual/draft invoice notifications

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1094]
In GitLab by @kdjstudios on Sep 12, 2018, 09:02

Hello Team,

I would like to bring up a reoccurring question that continues to be asked. It is about having multiple invoices in one billing cycle and that there is no way in the GUI that notates which invoice was a manual invoice and is leading to questions being asked and is taking up unnecessary amount of support time. 

My thoughts:
- We currently have a notification email when a manual draft invoice in created. This is only being sent to apperrors, it appears. 
 - We should add the username who created the draft be in the email and also it should go to the site's email and the user's email. 
 - We should add a notification email when the manual draft invoice is finalized and include the user name who did this and send to both site and user email in addition to apperrors.

- We should show or notate somehow in the gui that the invoice was manually created.
 - Billing history page (Maybe add a new column "Manual Invoice" and true or false result.)
 - Account Ledger (Maybe add a new column "Manual Invoice" and true or false result.)
 - What other places? Reports?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 12, 2018, 09:03

@smasih @pchaudhary @tim.traylor 

Please bring this up for discussion and if approved we can detail out further in order to get an accurate estimate.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 12, 2018, 09:03

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 26, 2019, 12:31

Marking into the New tickets to be reviewed.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 30, 2020, 04:19

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 30, 2020, 23:51

@kdjstudios @jredfield 

Thank you for the update.

> * We currently have a notification email when a manual draft invoice in created. This is only being sent to apperrors, it appears.

* As per the current functionality of SAB, we are sending an email to site email, user email,apperrors@sahosted.com and email notification for site managers.

> * We should add the username who created the draft be in the email and also it should go to the site's email and the user's email.

* We are storing the username who is created the manual incoice.

> * We should add a notification email when the manual draft invoice is finalized and include the user name who did this and send to both site and user email in addition to apperrors.

* We are not sending any notification when the manual invoice is finalized.

>* We should show or notate somehow in the gui that the invoice was manually created.

* We need to develop the functionality.

>* Billing history page (Maybe add a new column "Manual Invoice" and true or false result.)

* We need to develop the functionality.

>* Account Ledger (Maybe add a new column "Manual Invoice" and true or false result.)

* We need to develop the functionality.

>* What other places? Reports?
* I think we need to show everywhere where the invoices show.

***In short , we need to develop all functionality except first two points as mentioned on the top.**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 5, 2020, 13:17

@amishra 

* In short , we need to develop all functionality except first two points as mentioned on the top.**
> Thanks, Agreed. Lets get an estimate for adding this functionality. Please provide a detailed breakdown of all reports/screens that will be updated.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Feb 6, 2020, 05:31

@kdjstudios @jredfield 


Thank you for the update

> * We should add a notification email when the manual draft invoice is finalized and include the user name who did this and send to both site and user email in addition to apperrors.
 * `Estimate: less than 1 hrs`
> * Billing history page (Maybe add a new column "Manual Invoice" and true or false result.)
 * `Estimate: less than 30 min` 
> * Account Ledger (Maybe add a new column "Manual Invoice" and true or false result.)
 * `Estimate: less than 30 min`
> * What other places? Reports?
> * Please provide a detailed breakdown of all reports/screens that will be updated.
* First, we need to check all reports where the invoice is showing. It's will take around `2-3 hrs`. Once confirm all report we will post an estimate.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 6, 2020, 09:45

@amishra 

Thanks please post all estimates.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Feb 7, 2020, 03:10

@kdjstudios @jredfield 

Thank you for the update.

Please find the list of all reports which we need to add the column. Also please let us know where we need to put the column("Manual Invoice") on all the reports and same for account ledger & billing history page.

1. Detail Open AR
1. Invoice Journal
1. Payment Detail
1. Revenue Analysis Worksheet
1. Review Draft Invoices

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|3|Worked with Concepts, implementation ok|4|Low Impact|1.0|5|


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 7, 2020, 10:01

@jredfield 

Now that we have estimates, I think we can review this with Gary to determine if this should be a column or maybe an asterisk or what he would like to see to indicate that an invoice was a manually created one.

Below is a screenshot of the Invoice Journal report with a notation of how we would envision the manual invoice column to be shown. 

![Invoice_Journal](/uploads/f2444b4028c1dc587690ea5e97d4c8ae/Invoice_Journal.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 3, 2020, 09:47

This was reviewed with Gary and he would like for us to put this on the wish list until we have completed higher priority items. 

Gary would like for me to send an email to clarify the questions/confusion on the multiple invoices that have been reported. I will need to find all of the tickets by the users that have been submitted.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 1, 2020, 07:48

@kdjstudios @Robleski 

Thanks for the update.

We have completed development except following:-

*  Do we need to add changes for account ledger in client portal too. If yes then it will take around  30 minutes to add these changes.

*  In Detail Open AR Report we need to make a script and run it so that it will work for the historic data as well. That will take 2-3 hours.



Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2020, 08:50

@rranjan @Robleski 

Approved please proceed with the two additional updates above also.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 6, 2020, 07:37

@kdjstudios  @Robleski 

Thanks for the update.

* Ar Report has more than 7,00,000 records for the internal and more than 3,00,000 for the external. 
* Updating of all these records is taking more than three hours.
* So, we don't think it's a feasible step to take.
* We would suggest that for the historical data we should have "N/A" or "-". And the changes will reflect only on the reports that are created after its release.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 6, 2020, 14:39

@rranjan 

Do we not have a script/query that is able to do that? Why in the world would we be doing this manually, that seems crazy and illogical.

Also I am confused, are you starting there are 10 million manual invoices, or that there are a total of 10 million invoices total. As I would beleive there is probably only like 1000 of those are manual invoices in total ( of the 10 million).

@Robleski @aznbyte 

I am not the tech one to understand how long it does take to query and update those records, but I would say we should not spend more then 5 hours updating historical records, so if its going to be longer then that I would suggest we get Gary's approval to not do historical records and to only do future ones.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 7, 2020, 07:46

@kdjstudios  @Robleski 


> Do we not have a script/query that is able to do that? Why in the world would we be doing this manually, that seems crazy and illogical.

We are updating the record with the script only.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 11:18

@rranjan @Robleski 

I am confused, so if the script updates the records what is taking a user so long. we start the script and let it run then? what else is needed?

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 15, 2020, 06:53

@kdjstudios  @Robleski 


We have completed development with and setup branch on the merge server for your testing.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 15, 2020, 08:40

@rranjan @ranand @amishra 

May someone please answer my questions? http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1096#note_61084

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 15, 2020, 08:47

Marking for the weekend release

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 15, 2020, 09:09

@kdjstudios @Robleski 

@rranjan was trying to explain that we are creating a script to update the historical record in addition to add the manual column. 

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 20, 2020, 04:38

@kdjstudios @Robleski 

We have successfully released on both of the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 12, 2020, 11:37

Closing ticket. We have not received any notices that this is causing any issues.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 12, 2020, 11:37

closed
